### ISSUE TITLE
Error while fetching your routes !

### ISSUE THREAD

**Issue by targos** [Julien-R44/adonis-vscode-extension#12]
I don't know how to debug this. There is only one line in the logs:

```
TypeError: Cannot read properties of undefined (reading 'uri')
```

**Comment by targos** [comment]
Note that this error doesn't always happen. Yesterday my project opened with no errors and the routes were loaded correctly.

**Comment by Julien-R44** [comment]
I think it's because the command "ace list:routes --json" fails, but I can't see why.
Next time this happens, could you try clicking on this icon, to see if it changes anything? 

![image](https://user-images.githubusercontent.com/8337858/190861312-888b0f9d-2152-4336-b720-8f1e6e35da4b.png)

**Comment by targos** [comment]
Just got the error again!
Clicking refresh fixes it.

**Comment by targos** [comment]
It happens every time I open my project now. The error message is different though:
```
Error: Command failed: "node" ace list:routes --json
```

Is there anything I can do to help you debug this?

**Comment by Julien-R44** [comment]
I sometimes get the same error when I open my workspace. I'll try to investigate before the end of the week

**Comment by targos** [comment]
Today, the error message was "SyntaxError: Unexpected token e in JSON at position 0"

**Comment by RockyScube** [comment]
Error: Command failed: "node" ace list:routes --json
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'reflect-metadata'

**Comment by RockyScube** [comment]
@Julien-R44  Getting only this  in error:
`Error: Command failed: "node" ace list:routes --json`

**Comment by aayush123** [comment]
Is there any "debug" option on vscode extension so I can dig into this? I'd be happy to contribute a fix I can be pointed to where to look

**Comment by ralphdas** [comment]
Currently having a similar error: `SyntaxError: Expected ',' or ']' after array element in JSON at position 3`
Looks like its cannot parse the JSON output or the ```node ace list:routes --json``` command.

output of command:
```json
{
  "root": [
    {
      "domain": "root",
      "name": "drive.local.serve",
      "pattern": "/uploads/*",
      "methods": [
        "GET",
        "HEAD"
      ],
      "handler": "Closure",
      "middleware": []
    }, ...

**Comment by Julien-R44** [comment]
Hey, sorry I forgotten to mention this here, but, you can switch to the pre-release version of the extension, which normally fixes this issue!

( The pre-release supports both Adonis 5 and Adonis 6 )

**Comment by ralphdas** [comment]
Thanks for the update!

Op ma 13 nov 2023 om 19:07 schreef Julien Ripouteau <
***@***.***>:

**Comment by Julien-R44** [comment]
The extension has been released (officially, not a pre-release) as 1.2.0. Issue should be fixed
Let me know if it persists

**Comment by redeemefy** [comment]
I just got the same error
```shell
TypeError: Cannot read properties of undefined (reading 'uri')
```

I'm running VsCodium and this start happening after upgrading the OS.
```
Version: 1.88.1
Release: 24104
Commit: 0d01a72525276e5e3771fe5851b9c83de0de1e2b
Date: 2024-04-13T01:43:30.924Z (3 days ago)
Electron: 28.2.8
ElectronBuildId: undefined
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin arm64 23.4.0
```
